DictionaryConsoleApp/Program.cs

```diff
@@ -0,0 +1,159 @@
+﻿namespace DictionaryConsoleApp
+{
+    public class Program
+    {
+        private static void Main()
+        {
+            Exercise1();
+            Exercise2();
+            Exercise3();
+            Exercise4();
+        }
+
+        public static void Exercise1()
+        {
+            Console.WriteLine("Exercise 1.");
+
+            var employees = new Dictionary<Student, int>
+            {
+                { new Student(4, "John"), 1},
+                { new Student(5, "Henry"), 2 }
+            };
+
+            foreach (KeyValuePair<Student, int> pair in employees)
+            {
+                Console.WriteLine($"Student {pair.Key.Name}, estimation: {pair.Key.AverageRating}.");
+            }
+        }
+
+        public static void Exercise2()
+        {
+            var dictionaryEnglishFranch = new Dictionary<string, string>(10)
+            {
+                    { "book", "livre" },
+                    { "mouse", "souris" },
+                    { "dictionary", "dictionnaire" },
+                    { "notepad", "bloc-notes" },
+                    { "word", "mot" },
+                    { "time", "temps" },
+                    { "music", "musique" },
+                    { "game", "jeu" },
+                    { "flower", "fleur" },
+                    { "candles", "bougies" }
+            };
+
+            var input = Console.ReadLine();
+
+            if (!string.IsNullOrEmpty(input))
+            {
+                Console.WriteLine("Error. Repeat the input again.");
+                Exercise2();
+            }
+
+            Console.WriteLine(GetTranslation(input));
+
+            string GetTranslation(string input)
+            {
+                if (dictionaryEnglishFranch.ContainsKey(input))
+                {
+                    string translation;
+                    dictionaryEnglishFranch.TryGetValue(input, out translation);
+                    return translation;
```

Comment:
TryGetValue already contains the ContainsKey check
because it returns a boolean, you can be sure that if the value cannot be obtained, then it means that ContainsKey == false